### PR TITLE
PAT-2137 Deserialisation error in risk profiler

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,17 +1,9 @@
...
 }
 
-allOpen {
-  annotations(
-    "javax.persistence.Entity",
-    "javax.persistence.MappedSuperclass",
-    "javax.persistence.Embeddable"
-  )
-}
-
```
Comment (steverendell) [inline build.gradle.kts:14]: apparently the jpa plugin sorts this

File: build.gradle.kts [hunk 2]
```diff
@@ -64,10 +56,10 @@ dependencies {
   implementation("javax.activation:activation:1.1.1")
   implementation("javax.transaction:javax.transaction-api:1.3")
 
-  implementation("org.springdoc:springdoc-openapi-ui:1.6.3")
+  implementation("org.springdoc:springdoc-openapi-ui:1.6.5")
   implementation("org.springdoc:springdoc-openapi-kotlin:1.6.3")
   implementation("org.springdoc:springdoc-openapi-security:1.6.3")
-  // implementation("com.fasterxml.jackson.module:jackson-module-kotlin:2.13.1")
+  implementation("com.fasterxml.jackson.module:jackson-module-kotlin:2.13.1")
```
Comment (steverendell) [inline build.gradle.kts:62]: this is just effectively bumping up from 2.13.0

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/PollPrisonersService.kt [hunk 3]
```diff
@@ -84,9 +85,9 @@ class PollPrisonersService(
     val oldProfile: ProfileMessagePayload
     try {
       oldProfile = ProfileMessagePayload(
-        jacksonMapper.readValue(existing.escape, EscapeProfile::class.java),
-        jacksonMapper.readValue(existing.soc, SocProfile::class.java),
-        jacksonMapper.readValue(existing.violence, ViolenceProfile::class.java)
+        jacksonMapper.readValue(existing.escape),
+        jacksonMapper.readValue(existing.soc),
+        jacksonMapper.readValue(existing.violence)
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/PollPrisonersService.kt:90]: slightly simpler kotlin method invocation

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/PollPrisonersService.kt [hunk 4]
```diff
@@ -27,8 +29,7 @@ class PollPrisonersService(
   private val telemetryClient: TelemetryClient?,
   private val sqsService: SQSService
 ) {
-  private val jacksonMapper = ObjectMapper()
-  // private val defaultFromDateTime = LocalDateTime.of(2019, 10, 25, 0, 0)
+  private val jacksonMapper = ObjectMapper().registerModule(KotlinModule.Builder().build())
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/PollPrisonersService.kt:32]: This makes it aware of defaulted constructor arguments

File: src/test/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/PollPrisonersServiceTest.kt [hunk 5]
```diff
@@ -44,10 +46,15 @@ class PollPrisonersServiceTest {
...
+  private val ESCAPE_1 = EscapeProfile(
+    OFFENDER_1, "C", false,
+    true,
+    listOf(Alert(alertCode = "XER", dateCreated = LocalDate.parse("2016-01-13"), active = true, expired = false)),
+    null
+  )
   private val PROFILE_1 = PreviousProfile(
     OFFENDER_1,
-    """{"nomsId":"$OFFENDER_1","provisionalCategorisation":"C","activeEscapeList":false,"activeEscapeRisk":false,"riskType":"ESCAPE"}""",
+    """{"nomsId":"$OFFENDER_1","provisionalCategorisation":"C","activeEscapeList":false,"activeEscapeRisk":true,"escapeRiskAlerts":[{"alertCode":"XER","dateCreated":"2016-01-13","expired":false,"active":true}],"riskType":"ESCAPE"}""",
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/PollPrisonersServiceTest.kt:57]: added some actual Alerts !

**Review by connormaglynn — APPROVED** [review]
lgtm

**Pull request merged**